go-controller/pkg/ovn/ovn.go

```diff
@@ -1047,148 +1072,80 @@ func (oc *Controller) syncNodeGateway(node *kapi.Node, hostSubnets []*net.IPNet)
-					klog.Warningf(err.Error())
-				}
-				gatewaysFailed.Store(node.Name, true)
-			}
-
-			// ensure pods that already exist on this node have their logical ports created
-			options := metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector("spec.nodeName", node.Name).String()}
-			pods, err := oc.client.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), options)
-			if err != nil {
-				klog.Errorf("Unable to list existing pods on node: %s, existing pods on this node may not function")
-			} else {
-				oc.addRetryPods(pods.Items)
-				oc.requestRetryPods()
-			}
-
+			klog.Infof("Created Node: %s took: %v", node.Name, time.Since(start))
+			oc.retryNodes.checkAndDeleteRetryObj(node.Name, true)
 		},
 		UpdateFunc: func(old, new interface{}) {
 			oldNode := old.(*kapi.Node)
-			node := new.(*kapi.Node)
-
-			shouldUpdate, err := shouldUpdate(node, oldNode)
-			if err != nil {
-				klog.Errorf(err.Error())
-			}
-			if !shouldUpdate {
-				// the hostsubnet is not assigned by ovn-kubernetes
-				return
-			}
-
-			var hostSubnets []*net.IPNet
-			_, failed := addNodeFailed.Load(node.Name)
-			if failed {
-				hostSubnets, err = oc.addNode(node)
-				if err != nil {
-					klog.Errorf("NodeUpdate: error creating subnet for node %s: %v", node.Name, err)
+			newNode := new.(*kapi.Node)
+			oc.retryNodes.checkAndSkipRetryObj(oldNode.Name)
+			if retryEntry := oc.retryNodes.getObjRetryEntry(oldNode.Name); retryEntry != nil && retryEntry.oldObj != nil {
+				klog.Infof("Detected leftover old node during new node add  %s.", oldNode.Name)
+				if err := oc.deleteNodeObj(oldNode); err != nil {
+					oc.retryNodes.unSkipRetryObj(oldNode.Name)
+					klog.Errorf("Failed to delete stale node %s, during update: %v",
+						oldNode.Name, err)
 					return
 				}
-				addNodeFailed.Delete(node.Name)
-			}
-
-			_, failed = nodeClusterRouterPortFailed.Load(node.Name)
-			if failed || nodeChassisChanged(oldNode, node) || nodeSubnetChanged(oldNode, node) {
-				if err = oc.syncNodeClusterRouterPort(node, nil); err != nil {
-					if !util.IsAnnotationNotSetError(err) {
-						klog.Warningf(err.Error())
-					}
-					nodeClusterRouterPortFailed.Store(node.Name, true)
-				} else {
-					nodeClusterRouterPortFailed.Delete(node.Name)
-				}
-			}
-
-			_, failed = mgmtPortFailed.Load(node.Name)
-			if failed || macAddressChanged(oldNode, node) || nodeSubnetChanged(oldNode, node) {
-				err := oc.syncNodeManagementPort(node, hostSubnets)
-				if err != nil {
-					if !util.IsAnnotationNotSetError(err) {
-						klog.Errorf("Error updating management port for node %s: %v", node.Name, err)
-					}
-					mgmtPortFailed.Store(node.Name, true)
-				} else {
-					mgmtPortFailed.Delete(node.Name)
-				}
-			}
-
-			if nodeChassisChanged(oldNode, node) {
-				// delete stale chassis in SBDB if any
-				oc.deleteStaleNodeChassis(node)
+				// deletion was a success; remove delete retry entry
+				oc.retryNodes.removeDeleteFromRetryObj(newNode.Name)
```

Comment:
should we call it on `newNode.Name` or `oldNode.Name`? Here we've just successfully deleted `oldNode`.

Reply:
the remove is for newNode but all the logic above to see if we have stale obj should have been done for newNode not oldNode I renamed the var last night to make it clear but I incorrectly used them I will fix it